### PR TITLE
add reducers and actions

### PR DESCRIPTION
**Pull request by boo-shehab**
# Hi Reviewer 👋 

In this PR I added the reducers actions with all the requirements

### Project requirements
- Add Redux Toolkit(`npm install react-redux @reduxjs/toolkit`).
- Structure your application files using a "feature folder" approach and use the [*ducks* pattern](https://github.com/erikras/ducks-modular-redux) for your Redux files. Your files/folder structure could look like this:
    ```    
    ...
    /src
        |---/components
        |---/redux
            |--- /books
                    | books.js
            |--- /categories
                    | categories.js
            | configureStore.js
        | index.js
    ``` 
- Write your book's actions and reducer (in `/src/redux/books/books.js`) using the [*ducks* pattern](https://github.com/erikras/ducks-modular-redux). 
    - Define *action types* for adding and removing a book.
    - Set the initial state to be an empty array of books.
    - Export *Action Creators* for your *actions*.
    - Write your reducer and export it as default.
        - Define state changes for the actions that you created.
        - In case of unknown action - return the current state.
- Write your categories actions and reducer (in `/src/redux/categories/categories.js`) using the [*ducks* pattern](https://github.com/erikras/ducks-modular-redux). 
    - Define an *action type* for checking the status.
    - Set the initial state to be an empty array of categories.
    - Export *Action Creators* for your *actions*.
    - Write your reducer and export it as default.
        - For the check status action return a string "Under construction".
        - In case of unknown action - return the current state.

- Configure the Redux Store (`/src/redux/configureStore.js`):
    - Importing the necessary methods from Redux Toolkit.
    - Combine both reducers into a root reducer by using  `configureStore` function.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+const ADD_BOOK = 'ADD_BOOK';
+const REMOVE_BOOK = 'ERMOVE_BOOK';
```
Comment (rloterh) [inline src/redux/books/books.js]: - [ ] Going by the _duck pattern_, assigning values to _ADD_BOOK_ and _REMOVE_BOOK_ action types should be something like:
```ruby
       const ADD_BOOK = bookstore/books/ADD_BOOK;
```
        and
```ruby
      const REMOVE_BOOK = bookstore/books/REMOVE_BOOK;

```
Kindly ensure to improve upon these aspects.

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const CHECK_STATUES = 'CHECK_STATUES';
```
Comment (rloterh) [inline src/redux/categories/categories.js]: - [ ] Similarly, as per the _duck patterns approach_, assigning values to _CHECK_STATUS_ action type should be something like:
```ruby
      CHECK_STATUS = bookstore/categories/CHECK_STATUS;
```
 
Kindly ensure to improve upon these aspects.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Ahmed, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Good job implementing _duck pattern_ so far!
- Professional ReadMe ⭐ 

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

- [ ] OPTIONAL
Kindly replace the navigation link name `calculator` with `categories` accordingly, so as to correspond with the book categories content displayed on this page.
image

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @boo-shehab,

✔️ Implementation of ducks' pattern for action types

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**